.github/workflows/engine-and-cli.yml

```diff
@@ -25,7 +25,7 @@ jobs:
     uses: ./.github/workflows/_dagger_call.yml
     secrets: inherit
     with:
-      function: test all --race=true
+      function: test all --race=true --parallel=16
```

Comment:
Is this different from `NumCPU`s? I think we can leave this unset.

Reply:
Yes, we are explicitly setting this to `16` in CI.

The new CI runners usually have 48 and even 64 CPUs which makes tests fail. Before setting this, checks failed.